Missionframework/functions/fn_spawnVehicle.sqf

```diff
@@ -30,21 +30,27 @@ if (_classname isEqualTo "") exitWith {["Empty string given"] call BIS_fnc_error
 if (!canSuspend) exitWith {_this spawn KPLIB_fnc_spawnVehicle};
 
 private _newvehicle = objNull;
-private _spawnpos = zeropos;
+private _spawnpos = [];
 
 if (_precise) then {
     // Directly use given pos, if precise placement is true
     _spawnpos = _pos;
 } else {
     // Otherwise find a suitable position for vehicle spawning near given pos
     private _i = 0;
-    while {_spawnpos distance2d zeropos < 100} do {
+    while {_spawnPos isEqualTo []} do {
+        _i = _i + 1;
         _spawnpos = (_pos getPos [random 150, random 360]) findEmptyPosition [10, 100, 'B_Heli_Transport_01_F'];
```

Comment:
What about using `_classname` here? As far as I understand we use the Heli for infantry as we want place for whole group, however for vehicles we could use exact classname.
```suggestion
        _spawnpos = (_pos getPos [random 150, random 360]) findEmptyPosition [10, 100, _classname];
```